replica/database.cc

```diff
@@ -2014,6 +2018,7 @@ future<> database::do_apply(schema_ptr s, const frozen_mutation& m, tracing::tra
         auto& write_label = cf.get_rate_limiter_label_for_writes();
         auto token = dht::token::to_int64(dht::get_token(*s, m.key()));
         if (_rate_limiter.account_operation(write_label, token, table_limit, rate_limit_info) == db::rate_limiter::can_proceed::no) {
+            ++_stats->total_writes_rate_limited;
             co_await coroutine::return_exception(replica::rate_limit_exception());
         }
     }
```

Comment:
update_writes_failed's destructor will trigger, accounting for the failure.

Reply:
`update_write_metrics` also increases counters for rate limit exception and timeouts. I guess we don't care about the former because they shouldn't appear here, but what about timeouts?

Reply:
I think we should account for them here.

Reply:
(timeouts)